### ISSUE TITLE
License, documentation and other definition

### ISSUE THREAD

**Issue by kmubiin** [kmubiin/by2020#12]
Raised from #1 with revised text.

Some obvious things to explain in this issue:

A. What [license][1] to choose

* Single license for non-software project i.e. [Docs Project on Fedora Project Wiki (CC-BY-SA)][6], [FLOSS Manuals (GPL, formerly GFDL)][7];
* Dual-license for non-software project i.e. [Wikipedia (CC-BY-SA, GFDL)][8] and [alike (CC various)][9], except [Wikivoyage (CC-BY-SA)][10]; 

B. What is the work flow and branching strategy

* master should be at least separate the source data and files for GitHub Pages.

C. How to link between files in `docs/`

* Use [relative link][2] for local files
* Avoid absolute link whenever possible (troublesome to fix broken links)

D. What are the contributing guidelines

* Use uppercase letters like `LICENSE`
* Use [Markdown][3] like `README.md`
* Create [guideline][4] file `CONTRIBUTING.md` (may be placed under `.github/`)
* Create [issue template][5] file `ISSUE_TEMPLATE.md` (may be placed under `.github/`)

E. How to credit contributors and organize list (seriously)

* Simple: alphabetical order, one line per person (simple and fair, easiest to update)
* Detail: non-alphabetical, FIFO order, several predefined lines per person (the Linux project handles this way; update by themselves with proof?)
* Formatted detail: with Markdown formatting (useful list of names with roles and short description as seen on web pages, but difficult to update)

F. What to include in credit file for each contributor

* User name - The real name, nickname or both i.e. `John Doe (john99)`.
* User name reference - The email address, or link to public profile instead to pervent spam email, whichever that can verify user name is indeed refer to this user; This is to ensure the contribution with same user name is not claimable by someone else. Prefer link to public profile.
* User first contribution - The reference issue `#123` or reference commit `abcd456`.

Missing definitions that are not covered by previously created issues should be included in this issue. And that is all for defining issues for this project, I think.

  [1]: https://help.github.com/articles/adding-a-license-to-a-repository/
  [2]: https://github.com/blog/1395-relative-links-in-markup-files
  [3]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet
  [4]: https://help.github.com/articles/setting-guidelines-for-repository-contributors/
  [5]: https://help.github.com/articles/creating-an-issue-template-for-your-repository/
  [6]: https://fedoraproject.org/wiki/Docs_Project
  [7]: https://www.adamhyde.net/1198-2/
  [8]: https://en.wikipedia.org/wiki/Wikipedia:Copyrights
  [9]: https://commons.wikimedia.org/wiki/Commons:Licensing
  [10]: https://en.wikivoyage.org/wiki/Wikivoyage:Why_Wikivoyage_isn%27t_GFDL

**Comment by kmubiin** [comment]
The license is most likely same as Wikipedia, which is [Creative Commons Attribution ShareAlike][1]. The main reason is that this project may reuse some text from Wikipedia; Better be safe than sorry. Another supporting reason is same as [why Wikinews has switches the license][2].


  [1]: https://creativecommons.org/licenses/by-sa/4.0/
  [2]: https://en.wikinews.org/wiki/Wikinews_switches_to_Creative_Commons_license

**Comment by kmubiin** [comment]
The creation of some files i.e. `CONTRIBUTING.md`, `ISSUE_TEMPLATE.md` may have to be put on hold for now. I have to experience at first hand on event submission, so that issues may be identified first rather than writing guideline based on pure assumptions.

**Comment by kmubiin** [comment]
The files and directories in this repository may look like this:

    digest/
      |__ submit-event-result.csv
      |__ submit-event-eng.csv
      |__ submit-event-jpn.csv
      |__ submit-event-zsm.csv
      ...
    docs/
      |__ example/
      |      |__ event-blank.json
      |      |__ event-test.json
      |      ...
      |__ schema/
      |__ credits.md
      |__ faq.md
      |__ CONTRIBUTING.md
      |__ ISSUE_TEMPLATE.md
      ...
    japan/
      |__ 392.001-eng.json
      |__ 392.001-jpn.json
      |__ 392.001-zsm.json
      ...
    malaysia/
      |__ 458.001-eng.json
      |__ 458.001-jpn.json
      |__ 458.001-zsm.json
      ...
    world/ (instead of 'worldwide/')
      |__ 036.001-eng.json
      |__ 702.001-eng.json
      ...
      |__ 900.001-eng.json
      |__ 900.001-jpn.json
      |__ 900.001-zsm.json
      ...
    CREDITS
    LICENSE
    README.md

Top level files should be plain text and no redundant syntax, except for `README.md`. The top level file `CREDITS` is the default location to save list of contributors, whilst `docs/credits.md` may be created in alternative format i.e. Markdown.

More to be explained later.

**Comment by kmubiin** [comment]
The "manually rotated" is a straightforward way, but a mundane task to do.

I do not want to spend too much time to automate this task. My first internship back in 2009 was a good case study: I had spent most time to do data entries for hundreds of lines by hand, instead of spending time to learn how to write a script that has no guarantee to complete the job in time.

Things must have gotten better today. With my current knowledge and skills, there might be a better option to automate the task.

1. [Google Sheet API via Python][1] is one option, but the learning curve looks steep.
2. [Google Apps Script for Class Sheet][2] is another option, which uses JavaScript to write the script.

I will probably explore the latter, which is something similar to spreadsheet formula in Office Excel and LibreOffice Calc. I saw the possibility to implement "manually rotated" sheets by using method of `copyTo(spreadsheet)` and `clear(options)`.

Some relevant posts found:

* [Is there a way to automatically duplicate a large number of Google Sheets, name them, then copy their URLs to another sheet?][3], which hinted the method of `getRange()`.
* [Creating a copy of a sheet by copying values only?][4], which hinted the method of `getDataRange()` and `copyTo()` being used together.

But not sure how to trigger that periodically or by entries count. Need to research further.


  [1]: https://developers.google.com/sheets/api/quickstart/python
  [2]: https://developers.google.com/apps-script/reference/spreadsheet/sheet
  [3]: https://webapps.stackexchange.com/q/91067
  [4]: https://webapps.stackexchange.com/q/78255

**Comment by kmubiin** [comment]
Regarding use of trigger in Google Apps, I found the following resources:

* [Simple Triggers][1] on Google Apps Script;
* [Event Objects - Time-driven events][2] on Google Apps Script;
* [How to reset a Google Docs spreadsheet by script][3], which showing steps to achieve with mix of script code for email notification and web interface for time-driven events.
* [How to Schedule your Google Forms and Limit Submissions][4], which has simple steps and a likely usable code of Google Script.

Now I have found the documentation and several examples to clarify the usage, before attempting to try running the code. Much better than aimlessly doing trials and errors.


  [1]: https://developers.google.com/apps-script/guides/triggers/
  [2]: https://developers.google.com/apps-script/guides/triggers/events#time-driven_events
  [3]: https://blargepret.wordpress.com/2013/03/03/how-to-reset-a-google-docs-spreadsheet-by-script/
  [4]: https://www.labnol.org/internet/schedule-google-forms/20707/

**Comment by kmubiin** [comment]
My second thought: Instead of "manually rotated" submissions, sort by month and nodate categories. This is probably easier to implement than figuring out how to rotate files periodically by date.

**Comment by kmubiin** [comment]
I have recreated the event submission form on Google Forms.

Unlike previous forms, the idea is to use lowercase and short name for each question title, which will be saved as the column name in responses on Google Sheets. For explanation, use long text in description for each question. The responses file on Google Sheets is now much readable.

So what are remaining issues for the define milestone...?

**Comment by kmubiin** [comment]
Earlier, I wrote:

> The license is most likely same as Wikipedia, which is [Creative Commons Attribution ShareAlike][1]. The main reason is that this project may reuse some text from Wikipedia; Better be safe than sorry. Another supporting reason is same as [why Wikinews had switched the license][2].

That "reuse some text" is likely very less in amount in this project. As a result, that reuse may be permissible under "fair use".

Unlike Wikipedia and Wikinews, this project is not intending to be in form of full-text article but rather in form of limited data set. Licensing under ShareAlike may reduce the reusablility of data given the explicit condition to give attribution and such.

In terms of crediting contributors, I have already included in data structure of JSON, which is defined at "event">"meta">"submit_author". Although there is no obligation to giving credit to Public Domain materials, [nevertheless, it is “courteous” to cite the source and/or provide a credit][3].

That being said, this project might be licensed under CC0 (Public Domain).


  [1]: https://creativecommons.org/licenses/by-sa/4.0/
  [2]: https://en.wikinews.org/wiki/Wikinews_switches_to_Creative_Commons_license
  [3]: http://www.guidethroughthelegaljungleblog.com/2008/07/giving-credit-for-public-domain-and-fair-use-materials.html
